### PR TITLE
test: add test for linearizability of unmaterialized sources

### PR DESCRIPTION
**Pull request by benesch**
Add a test that demonstrates the most obvious linearizability defect in
unmaterialized sources: that they don't promise to show you anything in
particular when you query them.

See https://github.com/MaterializeInc/materialize/issues/2310, https://github.com/MaterializeInc/materialize/issues/10587, https://github.com/MaterializeInc/materialize/issues/11048, and https://github.com/MaterializeInc/materialize/issues/11072.

### Motivation

  * This PR ~fixes~ adds a test for a recognized bug.

### Testing

- [x] This PR has adequate test coverage / QA involvement has been duly considered.

### Release notes

This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):

  - TODO: if we fix the bug, maybe write a release note?

### CODE REVIEW

File: test/testdrive/unmaterialized-sources.td [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+$ kafka-ingest format=bytes topic=data repeat=100000
+1
+
+> CREATE SOURCE kafka
+  FROM KAFKA BROKER '${testdrive.kafka-addr}' TOPIC 'testdrive-data-${testdrive.seed}'
+  FORMAT TEXT
+
+> SELECT text FROM kafka
+100000
```
Comment (frankmcsherry) [inline test/testdrive/unmaterialized-sources.td:52]: Should this be `SELECT count(*)`?

**Review by frankmcsherry — APPROVED** [review]
I like what this tests, but I do not understand how we commit the test / initiate the process of making things that fail this test not land?

**Comment by benesch** [comment]
Folding into #11192.

File: test/testdrive/unmaterialized-sources.td (hunk 1 above)
Comment (benesch) [reply test/testdrive/unmaterialized-sources.td:52]: Oops, yes. Thanks. Fixed in #11192.

**Pull request closed without merging**